File: app/src/main/java/com/example/configurers/Analytics.java
package com.example.configurers;

import android.app.Application;
import android.support.annotation.NonNull;

import com.example.BuildConfig;
import com.google.firebase.analytics.FirebaseAnalytics;

import com.example.analytics.AnalyticsActivityLifecycle;

public final class Analytics {
	public static void setup(@NonNull Application application) {
		application.registerActivityLifecycleCallbacks(new AnalyticsActivityLifecycle());
		FirebaseAnalytics.getInstance(application).setAnalyticsCollectionEnabled(!BuildConfig.DEBUG);
	}
}

File: app/src/main/kotlin/com/example/configurers/Analytics.kt
package com.example.configurers

import android.app.Application
import com.example.BuildConfig
import com.example.analytics.AnalyticsActivityLifecycle
import com.google.firebase.analytics.FirebaseAnalytics

object Analytics {
  fun setup(application: Application) {
    application.registerActivityLifecycleCallbacks(AnalyticsActivityLifecycle())
    FirebaseAnalytics.getInstance(application).setAnalyticsCollectionEnabled(!BuildConfig.DEBUG)
  }
}
